docs/introduction/Installation.md

```diff
@@ -8,6 +8,85 @@
 
 ## vcpkg
 
+## Compiling from Source - Ubuntu 16.04 LTS (Xenial Xerus) 
+Please run the following commands and read the comments as you go.
+You will notice that we update and upgrade quite frequently, this is to ensure that all dependancies are met at all times. You will also notice that we are installing some libraries from source, there are two reasons for this; firstly in the event that the library is not available through apt-get or secondly, because we want the very latest version (not the version available in apt-get).
+```
+sudo apt-get update
+sudo apt-get upgrade
+sudo apt-get install cmake
+sudo apt-get install zlibc
+sudo apt-get install zlib1g
+sudo apt-get install zlib1g-dev
+sudo apt-get install libssl-dev
+sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+sudo apt update
+sudo apt-get upgrade
+sudo apt-get install gcc-6 g++-6
```

Comment:
This is another good question, I am not sure if that would work, I would need to test that.
I can confirm however, that the users can always upgrade to the baseline version of GCC 6 through the use of APT (which takes care of dependencies up until GCC 6.1). From there if the users specify the GCC 6 environment variables (as per my documentation)  the new 6.3.0 will install from source without any issues.